TestFoundation/TestFileHandle.swift

```diff
@@ -279,46 +280,47 @@ class TestFileHandle : XCTestCase {
     }
 
     func test_truncateFile() {
-        mkstemp(template: "test_truncateFile.XXXXXX") { (fh) in
-            fh.truncateFile(atOffset: 50)
-            XCTAssertEqual(fh.offsetInFile, 50)
+        let (fd, path) = try! _NSCreateTemporaryFile("test_truncate")
```

Comment:
This test depends on internal calls and thus must be guarded with `NS_FOUNDATION_ALLOWS_TESTABLE_IMPORT`.

Reply:
Alternatively you could use `NSTemporaryDirectory()` to get a tmp dir name then use something like:
```
let url = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(ProcessInfo.processInfo.globallyUniqueString)
let fh = try FileHandle(forWritingTo: url)
```

Also, make `func test_truncateFile() throws` and let the throws propogate instead of using `try!`